### PR TITLE
Black

### PR DESCRIPTION
**Pull request by jl-wynen**
Check out the first commit to see the config. The 2nd applies black via `pre-commit run -a`.

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -17,13 +17,11 @@ repos:
...
-    rev: v0.32.0
+  - repo: https://github.com/psf/black
+    rev: 23.1.0
     hooks:
-      - id: yapf
-        args: [ "-i", "-r" ]
-        types: [ "python" ]
-        additional_dependencies: [ "toml" ]
+      - id: black
+        exclude: src/scipp/core/cpp_classes.pyi
```
Comment (SimonHeybrock) [inline .pre-commit-config.yaml:24]: Is this not using the `pyproject.toml`?

**Review by YooSunYoung — APPROVED** [review]

File: .pre-commit-config.yaml (hunk 1 above)
Comment (jl-wynen) [reply .pre-commit-config.yaml:24]: I tried to use `exclude`, `extend_exclude` and `force_exclude` in pyroject.toml. But it seems that pre-commit calls it in such a way that it ignores those settings.

**Comment by jl-wynen** [comment]
Added a file to ignore revs in git blame as per https://stackoverflow.com/questions/53502654/how-do-i-run-a-code-formatter-over-my-source-without-modifying-git-history/69118451#69118451

But I think that everyone needs to apply this config for their clone.

**Comment by SimonHeybrock** [comment]
> But I think that everyone needs to apply this config for their clone.

Do mean you, run the following:
```
git config blame.ignoreRevsFile .git-blame-ignore-revs
```
?

**Review by SimonHeybrock — APPROVED** [review]

File: .git-blame-ignore-revs [hunk 2]
```diff
@@ -0,0 +1,2 @@
+# Run black on .py files
+f7e54a7ca95468b0b751b93931dc64317ccef44b
```
[the remark this answers is not in the document]
Comment (jl-wynen) [reply .git-blame-ignore-revs]: I rebased the branch. Wasn't aware that that changes commit hashes.

**Comment by jl-wynen** [comment]
> > But I think that everyone needs to apply this config for their clone.
> 
> Do mean you, run the following:
> 
> ```
> git config blame.ignoreRevsFile .git-blame-ignore-revs
> ```
> 
> ?

Yes

**Pull request merged**